File: scalechain-codec-proto/src/test/scala/io/scalechain/blockchain/proto/codec/CodecTestUtil.scala
package io.scalechain.blockchain.proto.codec

import io.scalechain.blockchain.proto.ProtocolMessage
import scodec.Codec
import scodec.bits.BitVector


trait CodecTestUtil {
  def decodeFully[T](bitVector:BitVector)(implicit codec:Codec[T]) : T =
    codec.decode(bitVector).require.value

  def encode[T](message : T)(implicit codec:Codec[T]) =
    codec.encode(message).require
}

File: scalechain-proto-codec/src/main/kotlin/io/scalechain/blockchain/proto/codec/CodecTestUtil.kt
package io.scalechain.blockchain.proto.codec

interface CodecTestUtil {
  fun<T> decodeFully(codec:Codec<T>, bytes : ByteArray) : T = codec.decode(bytes)!!
  fun<T> encode(codec:Codec<T>, message : T) : ByteArray = codec.encode(message)
  fun<T> roundTrip(codec:Codec<T>, message : T) : Boolean {
    return codec.decode(codec.encode(message)) == message
  }
}

